src/main/java/io/github/mdcteam/practice/items/ItemBasicArmor.java

```diff
@@ -0,0 +1,50 @@
+package io.github.mdcteam.practice.items;
+
+import com.google.common.collect.Sets;
+import io.github.mdcteam.practice.PracticeMod;
+import net.minecraft.init.SoundEvents;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.ItemArmor;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.SoundEvent;
+import net.minecraftforge.common.util.EnumHelper;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class ItemBasicArmor extends ItemArmor {
+
+    public static Set<ItemBasicArmor> createArmorSet(String name, int durability, int[] reductions, int enchantability, float toughness, SoundEvent equipSound, EntityEquipmentSlot... equipmentSlots) {
+        ResourceLocation resLoc = new ResourceLocation(PracticeMod.MOD_ID, name);
+        return createArmorSet(name, EnumHelper.addArmorMaterial(resLoc + "_material", resLoc.toString(), durability, reductions, enchantability, equipSound, toughness), equipmentSlots);
+    }
+
+    public static Set<ItemBasicArmor> createArmorSet(String name, ArmorMaterial material, EntityEquipmentSlot... equipmentSlots) {
+        Set<ItemBasicArmor> armorSet = new HashSet<>();
+        if(equipmentSlots.length == 0) equipmentSlots = new EntityEquipmentSlot[] {EntityEquipmentSlot.HEAD, EntityEquipmentSlot.CHEST, EntityEquipmentSlot.LEGS, EntityEquipmentSlot.FEET};
+        Set<EntityEquipmentSlot> slotSet = Sets.newHashSet(equipmentSlots);
+        slotSet.forEach(slot -> {
+            if (slot.getSlotType() != EntityEquipmentSlot.Type.ARMOR)
+                throw new IllegalArgumentException("Cannot create armour from the equipment slot type " + slot);
```

Comment:
Is this exception caught and handled anywhere or does minecraft/liblib/java do that implicitly and am I just super rusty?

Reply:
I believe that MC/Forge will catch it and create an error report for errors on the main threads.
Tbh it should never be thrown - just using it to make sure people don't do silly things :)